### PR TITLE
Upgrade project to Godot 4 & fix header regex bug

### PR DESCRIPTION
**Pull request by meaf75**
Upgrades the project to godot 4 and fixes header regex

# The Problem
The _handle_request method was placing body fields inside the headers object

# Solution
Replace the regex for the headers ("^(?<key>[^:]+): (?<value>.+)$" → "^(?<key>[\\w-]+): (?<value>(.*))$")

# Additions
- Added example main_scene node to the project with an script that starts the server and other that acts like HttpRouter
- Added method "get_body_parsed" to the "HttpRequest" that returns the body into an object doing operations like if the content-type of the request is "application/json" return the raw_body parsed using the JSON.parse function, if the content-type of the request is "application/x-www-form-urlencoded" then split the body string into parts and build the body object

### CODE REVIEW

File: addons/godottpd/http_response.gd [hunk 1]
```diff
@@ -44,24 +44,24 @@ func send_raw(status_code: int, data: PoolByteArray = [], content_type: String =
...
+	send(status_code, JSON.stringify(data), "application/json")
 
 
 # Sets the response’s header "field" to "value"
 #
 # #### Parameters
 # - field: the name of the header i.e. "Accept-Type"
 # - value: the value of this header i.e. "application/json"
-func set(field: String, value: String) -> void:
+func set(field: StringName, value: Variant) -> void:
```
Comment (dploeger) [inline addons/godottpd/http_response.gd:64]: Why Variant? Can a http header be something else than a string?

File: addons/godottpd/http_response.gd [hunk 2]
```diff
@@ -70,18 +70,18 @@ func set(field: String, value: String) -> void:
...
-#			 for this specific cookie, in the { "secure" : "true" } format 
+# - options: a Dictionary of ![cookie attributes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#attributes)
+#			 for this specific cookie, in the { "secure" : "true" } format
 func cookie(name: String, value: String, options: Dictionary = {}) -> void:
 	var cookie: String = name+"="+value
 	if options.has("domain"): cookie+="; Domain="+options["domain"]
 	if options.has("max-age"): cookie+="; Max-Age="+options["max-age"]
 	if options.has("expires"): cookie+="; Expires="+options["expires"]
-	if options.has("path"): cookie+="; Path="+options["path"]
+	if options.has("path"): cookie+="; Path3D="+options["path"]
```
Comment (dploeger) [inline addons/godottpd/http_response.gd]: I guess that's a replacement error, right? It should be Path

File: addons/godottpd/http_response.gd [hunk 3]
```diff
@@ -70,18 +70,18 @@ func set(field: String, value: String) -> void:
...
 	if options.has("secure"): cookie+="; Secure="+options["secure"]
 	if options.has("httpOnly"): cookie+="; HttpOnly="+options["httpOnly"]
-	if options.has("path"): cookie+="; Path="+options["path"]
-	if options.has("sameSite"): 
+	if options.has("path"): cookie+="; Path3D="+options["path"]
```
Comment (dploeger) [inline addons/godottpd/http_response.gd]: Same here

**Review by dploeger — CHANGES_REQUESTED** [review]
Sorry for not getting back to you sooner.

Thanks a lot for this! ❤️ 

Some smaller remarks only.

File: addons/godottpd/http_response.gd (hunk 1 above)
Comment (meaf75) [reply addons/godottpd/http_response.gd:64]: the "set" function of this class is overriding the "set" function from the class "Object" which is defined as:
`void set(property: StringName, value: Variant)`

Comment (dploeger) [reply addons/godottpd/http_response.gd:64]: Ohhh!

**Pull request merged**